compaction/compaction_manager.cc

```diff
@@ -683,10 +688,22 @@ sstables_task_executor::~sstables_task_executor() {
     _cm._stats.pending_tasks -= _sstables.size() - (_state == state::pending);
 }
 
+void compaction_task_executor::fail_exceptionally(std::exception_ptr ex) noexcept {
+    if (!_compaction_done.available()) {
+        _compaction_done_promise.set_exception(std::move(ex));
+    }
+}
+
 future<compaction_manager::compaction_stats_opt> compaction_task_executor::run_compaction() noexcept {
     try {
-        _compaction_done = do_run();
-        return compaction_done();
+        return do_run().then_wrapped([&] (auto f) {
+            if (f.failed()) {
```

Comment:
I wonder if it's safe to call here `forward_to(std::move(_compaction_done_promise))`